docs/oathkeeper/guides/websockets.mdx

```diff
@@ -0,0 +1,359 @@
+:::
+
+Let's create a simple echo WebSocket service that sends back an
+accepted message. We'll use Go, Gin to build our application,
+and Ory Kratos to handle user login/signup and verification flows.
+
+## Installing Ory Kratos and Ory Oathkeeper
+
+You can create any directory for the experiments and create `docker-compose.yml` file with the following content
+
+<CodeBlock className="language-yaml">{`version: "3.7"
+services:
+  oathkeeper:
+    image: oryd/oathkeeper:${useLatestRelease('oathkeeper')}
+    depends_on:
+      - kratos
+    ports:
+      - 8080:4455
+      - 4456:4456
+    command:
+      serve proxy -c "/etc/config/oathkeeper/oathkeeper.yml"
+    environment:
+      - LOG_LEVEL=debug
+    restart: on-failure
+    networks:
+      - intranet
+    volumes:
+      - ./oathkeeper:/etc/config/oathkeeper
+  postgres-kratos:
+    image: postgres:9.6
+    environment:
+      - POSTGRES_USER=kratos
+      - POSTGRES_PASSWORD=secret
+      - POSTGRES_DB=kratos
+    networks:
+      - intranet
+  kratos-migrate:
+    image: oryd/kratos:${useLatestRelease('kratos')}
+    links:
+      - postgres-kratos:postgres-kratos
+    environment:
+      - DSN=postgres://kratos:secret@postgres-kratos:5432/kratos?sslmode=disable&max_conns=20&max_idle_conns=4
+    networks:
+      - intranet
+    volumes:
+      - type: bind
+        source: ./kratos
+        target: /etc/config/kratos
+    command: -c /etc/config/kratos/kratos.yml migrate sql -e --yes
+  kratos:
+    image: oryd/kratos:${useLatestRelease('kratos')}
+    links:
+      - postgres-kratos:postgres-kratos
+    environment:
+      - DSN=postgres://kratos:secret@postgres-kratos:5432/kratos?sslmode=disable&max_conns=20&max_idle_conns=4
+    ports:
+      - '4433:4433'
+      - '4434:4434'
+    volumes:
+      - type: bind
+        source: ./kratos
+        target: /etc/config/kratos
+    networks:
+      - intranet
+    command: serve -c /etc/config/kratos/kratos.yml --dev --watch-courier
+  kratos-selfservice-ui-node:
+    image: oryd/kratos-selfservice-ui-node:latest
+    environment:
+      - KRATOS_PUBLIC_URL=http://kratos:4433/
+      - KRATOS_BROWSER_URL=http://127.0.0.1:4433/
+    networks:
+      - intranet
+    ports:
+      - "4455:3000"
+    restart: on-failure
+  mailslurper:
+    image: oryd/mailslurper:latest-smtps
+    ports:
+      - '4436:4436'
+      - '4437:4437'
+    networks:
+      - intranet
+networks:
+  intranet:`}</CodeBlock>
+
+This example uses the following network architecture
+
+- 4433 port is a public ("Browser") API of Ory Kratos
+- 4434 is an admin API of Ory Kratos
+- 4455 is a port of the User Interface implemented by self-service-ui
+- 8080 is a port of Ory Oathkeeper
+
+The rest ports or services will be available only inside the internal network.
+
+## Configuring Ory Oathkeeper and Ory Kratos
```

Comment:
You're describing steps here. Maybe it'd be a good idea to put this in an ordered list structure? You could get rid of "and then's", etc. this way and focus on describing the steps users should take.

Reply:
List structures have rendering issues when mixed with configuration blocks.

Reply:
If you're having problems implementing that, maybe have a look at my PR? I use codeblocks in lists here and I didn't really have any problems with rendering. 

https://github.com/ory/docs/pull/779